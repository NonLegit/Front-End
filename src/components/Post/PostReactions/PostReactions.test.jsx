import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@mui/system';
import { screen, render, fireEvent } from '@testing-library/react';
import PostReactions from './PostReactions';
import theme from '../../../styles/theme';
import EditPostContextProvider from '../../../contexts/EditPostContext';
import HiddenPostsContextProvider from '../../../contexts/HiddenPostsContext';

describe(PostReactions, () => {
  it('PostReactions renders correctly', () => {
    const testCase = {
      matchSm: true, comments: 100, matchMd: true, votes: 10,
    };
    const tree = renderer
      .create(
        <Router>
          <ThemeProvider theme={theme}>
            <HiddenPostsContextProvider>
              <EditPostContextProvider>
                <PostReactions
                  votes={testCase.votes}
                  matchSm={testCase.matchSm}
                  matchMd={testCase.matchMd}
                  comments={testCase.comments}
                />
              </EditPostContextProvider>
            </HiddenPostsContextProvider>

          </ThemeProvider>
        </Router>
        ,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Save button works correctly', () => {
    const testCase = {
      matchSm: true, comments: 100, matchMd: true, votes: 10,
    };
    render(
      <Router>
        <ThemeProvider theme={theme}>
          <HiddenPostsContextProvider>
            <EditPostContextProvider>
              <PostReactions
                votes={testCase.votes}
                matchSm={testCase.matchSm}
                matchMd={testCase.matchMd}
                comments={testCase.comments}
              />
            </EditPostContextProvider>
          </HiddenPostsContextProvider>

        </ThemeProvider>
      </Router>,
    );
    expect(screen.queryByText('Save')).toBeInTheDocument();
    expect(screen.queryByText('Unsave')).not.toBeInTheDocument();
    const saveButton = screen.getByTestId('save button');
    fireEvent.click(saveButton);
    expect(screen.queryByText('Unsave')).toBeInTheDocument();
    expect(screen.queryByText('Save')).not.toBeInTheDocument();
  });
});
