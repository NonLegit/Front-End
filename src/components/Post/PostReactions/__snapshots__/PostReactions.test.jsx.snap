// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostReactions PostReactions renders correctly 1`] = `
<div
  className="MuiBox-root css-1iio7w4"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textThird MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textThird MuiButton-sizeMedium MuiButton-textSizeMedium css-4lmdq8-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ChatBubbleOutlineRoundedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4v12H5.17L4 17.17V4h16m0-2H4c-1.1 0-2 .9-2 2v15.59c0 .89 1.08 1.34 1.71.71L6 18h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2z"
        />
      </svg>
    </span>
    100
     
    comments
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textThird MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textThird MuiButton-sizeMedium MuiButton-textSizeMedium css-4lmdq8-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ShareOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92s2.92-1.31 2.92-2.92c0-1.61-1.31-2.92-2.92-2.92zM18 4c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zM6 13c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm12 7.02c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
        />
      </svg>
    </span>
    Share
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textThird MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textThird MuiButton-sizeMedium MuiButton-textSizeMedium css-4lmdq8-MuiButtonBase-root-MuiButton-root"
    data-testid="save button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hk6mdv-MuiSvgIcon-root"
        data-testid="BookmarkBorderIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2zm0 15-5-2.18L7 18V5h10v13z"
        />
      </svg>
    </span>
    Save
  </button>
  <div
    className="MuiBox-root css-hyxlzm"
    onClick={[Function]}
    onTouchEnd={[Function]}
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorThird MuiIconButton-sizeMedium css-1mr5kkq-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MoreHorizOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </button>
    <ul
      className="MuiList-root MuiList-padding css-kaynf2-MuiList-root"
      display="none"
      match={false}
    >
      <div
        className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-18fvm0w-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ModeEditOutlineOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 21h3.75L17.81 9.94l-3.75-3.75L3 17.25V21zm2-2.92 9.06-9.06.92.92L5.92 19H5v-.92zM18.37 3.29a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-iw86h9-MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            edit post
          </span>
        </div>
      </div>
      <hr
        className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-18fvm0w-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityOffOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 6c3.79 0 7.17 2.13 8.82 5.5-.59 1.22-1.42 2.27-2.41 3.12l1.41 1.41c1.39-1.23 2.49-2.77 3.18-4.53C21.27 7.11 17 4 12 4c-1.27 0-2.49.2-3.64.57l1.65 1.65C10.66 6.09 11.32 6 12 6zm-1.07 1.14L13 9.21c.57.25 1.03.71 1.28 1.28l2.07 2.07c.08-.34.14-.7.14-1.07C16.5 9.01 14.48 7 12 7c-.37 0-.72.05-1.07.14zM2.01 3.87l2.68 2.68C3.06 7.83 1.77 9.53 1 11.5 2.73 15.89 7 19 12 19c1.52 0 2.98-.29 4.32-.82l3.42 3.42 1.41-1.41L3.42 2.45 2.01 3.87zm7.5 7.5 2.61 2.61c-.04.01-.08.02-.12.02-1.38 0-2.5-1.12-2.5-2.5 0-.05.01-.08.01-.13zm-3.4-3.4 1.75 1.75c-.23.55-.36 1.15-.36 1.78 0 2.48 2.02 4.5 4.5 4.5.63 0 1.23-.13 1.77-.36l.98.98c-.88.24-1.8.38-2.75.38-3.79 0-7.17-2.13-8.82-5.5.7-1.43 1.72-2.61 2.93-3.53z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-iw86h9-MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            hide
          </span>
        </div>
      </div>
      <hr
        className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-18fvm0w-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteOutlineOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4h-3.5z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-iw86h9-MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            delete
          </span>
        </div>
      </div>
      <hr
        className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-18fvm0w-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FlagOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12.36 6 .4 2H18v6h-3.36l-.4-2H7V6h5.36M14 4H5v17h2v-7h5.6l.4 2h7V6h-5.6L14 4z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-iw86h9-MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            report
          </span>
        </div>
      </div>
    </ul>
  </div>
</div>
`;
